go-controller/pkg/ovn/gateway.go

```diff
@@ -804,7 +804,7 @@ func (gw *GatewayManager) GatewayInit(
 		if err != nil {
 			return fmt.Errorf("failed to update SNAT rule for pod on router %s error: %v", gatewayRouter, err)
 		}
-	} else {
+	} else if gw.netInfo.IsDefault() { // only delete the subnet SNAT for default network not for P-UDNs or S-UDNs
```

Comment:
This else should be left alone as well. If in the previous `if` we determined that we don't need the SNATs, then the only alternative is to delete them. There is no room for `I don't need the SNAT but I don't delete them either`. Add them or delete them, but don't do nothing. Level driven.

I don't know though if the delete might be deleting something unexpected if it is too broad. But I guess in that case the delete should be fixed instead.

Reply:
this check can be left alone yes, it was only a precautionary check I added but I don't think its necessary. I will remove this.